pkg/daemon/update.go

```diff
@@ -788,6 +788,25 @@ func (dn *Daemon) deleteStaleData(oldConfig, newConfig *mcfgv1.MachineConfig) er
 				}
 				glog.V(2).Infof("Removing file %q completely", f.Path)
 			} else if _, err := os.Stat(origFileName(f.Path)); err == nil {
+				// Add a check for backwards compatibility: basically if the file doesn't exist in /usr/etc
+				// and no rpm is claming it, we assume that the orig file came from a wrongful backup of a
+				// MachineConfig file instead of a RHCOS file. See https://bugzilla.redhat.com/show_bug.cgi?id=1814397
+				if _, err := exec.Command("rpm", "-qf", f.Path).CombinedOutput(); err != nil {
+					if err := os.Remove(origFileName(f.Path)); err != nil {
+						return errors.Wrapf(err, "deleting orig file %q: %v", origFileName(f.Path), err)
+					}
+					continue
```

Comment:
instead of continuing, should we go ahead at the routine that deletes the file (at https://github.com/openshift/machine-config-operator/pull/1593/commits/7a68edd2d2b87ea481fde86f7c3a5c849565bea4#diff-06961b075f1753956d802ba954d2cfb5R819)